src/DynamoCoreWpf/Rendering/HelixRenderPackage.cs

```diff
@@ -555,12 +521,281 @@ public IEnumerable<int> PointIndices
             get { return points.Indices.ToArray(); }
         }
 
+        [Obsolete("Do not use! Use the methods in IRenderPackageSupplement to add mesh texture maps.")]
         public IEnumerable<byte> Colors
         {
             get { return colors; }
         }
 
-        public int ColorsStride { get; set; }
+        [Obsolete("Do not use! Use the methods in IRenderPackageSupplement to add mesh texture maps.")]
+        public int ColorsStride
+        {
+            get => colorStride;
+            set
+            {
+                if (!AllowLegacyColorOperations)
+                {
+                    throw new LegacyRenderPackageMethodException();
+                }
+                colorStride = value;
+            }
+        }
+
+        #endregion
+
+        #region IRenderLabels implementation
+
+        ///// <summary>
+        ///// Get label data; label string and associated position
+        ///// </summary>
+        public List<Tuple<string, float[]>> LabelData
```

Comment:
Can we just have one property - either `LabelPlaces` or `LabelData` instead of having both? I think they are essentially the same data, just formatted differently, I'm I'm not wrong?